### PR TITLE
Add new accessor function getByKeys()

### LINKED ISSUE #28: New accessor function to retrieve a list of values by keys
New accessor function to retrieve a list of values by keys.

**Example:**
```js
const a = 1;
const b = 2;
const sum = a + b;

store.set({ a, b });
store.set({ sum });

const contents = store.getByKeys(['a', 'b', 'sum']);
console.log(contents);
// > [1, 2, 3]

...
const [a, b, sum] = store.getByKeys(['a', 'b', 'sum']);
console.log({ a, b, sum });

// > { a: 1, b: 2, sum: 3 }
```

### PR DESCRIPTION
**Pull request by spencerwi**
Added the `getByKeys()` method described in #28, and exercised it in some tests (and updated the README). Y'all's changelog appears autogenerated, so I didn't touch that, as a heads-up.

Sorry for the multiple "referenced a commit" over on the issue -- to keep things clean and down to a single commit, I kept force-pushing to my own fork repo (normally bad form, I know, but for a quick PR branch I figured it seemed alright).

Fixes https://github.com/leapfrogtechnology/async-store/issues/28

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
@spencerwi Awesome. Thanks for contributing :1st_place_medal: :+1: :pray:.

There are few changes I'd request before signing off: 
1. Tweak the implementation for `getByKeys` - to return an identical "array" of values instead of an object. Please check the example I created in the issue https://github.com/leapfrogtechnology/async-store/issues/28.

   This is the desired behavior:
   ```diff
   - const results = store.getByKeys(['a', 'b', 'sum']); // {a: 1, b: 2, sum: 3}
   + const results = store.getByKeys(['a', 'b', 'sum']); // [1, 2, 3]
   ``` 
   This gives you an ability to do this more intuitively:
   ```ts
   const [a, b, sum] = store.getByKeys(['a', 'b', 'sum']);
   ```
2. Update tests, docblocks and README as per the above change.

File: README.md [hunk 1]
```diff
@@ -178,6 +178,13 @@ Gets a value by a key from the store.
 const foo = store.get('foo');
 ```
 
+### getByKeys()
+
+Gets multiple values by key from the store, and returns an object with those keys and their corresponding values.
```
Comment (kabirbaidhya) [inline README.md]: Update this.

File: src/index.ts [hunk 2]
```diff
@@ -136,6 +136,25 @@ export function get(key: string): any {
...
+ * Retrieves all values that correspond to a given list of keys.
+ *
+ * Example:
+ *  const a = 1;
+ *  const b = 2;
+ *  const sum = a + b;
+ *  store.set({a, b, sum})
+ *  const results = store.getByKeys(['a', 'b', 'sum']); // {a: 1, b: 2, sum: 3}
+ *
+ * @param {Array<string>} keys
```
Comment (kabirbaidhya) [inline src/index.ts]:
```suggestion
 * @param {string[]} keys
```

File: src/index.ts [hunk 3]
```diff
@@ -136,6 +136,25 @@ export function get(key: string): any {
...
+ * @returns {*}
+ */
+export function getByKeys(keys: string[]): any {
```
Comment (kabirbaidhya) [inline src/index.ts]: You can make it a generic function to have a specific return type. 
```suggestion
export function getByKeys<T>(keys: string[]): T[] {
```

File: src/impl/domain.ts [hunk 4]
```diff
@@ -118,6 +118,28 @@ export function get(key: string): any {
...
+ *  const a = 1;
+ *  const b = 2;
+ *  const sum = a + b;
+ *  store.set({a, b, sum})
+ *  const results = store.getByKeys(['a', 'b', 'sum']); // {a: 1, b: 2, sum: 3}
+ *
+ * @param {Array<string>} keys
+ * @returns {*}
+ */
+export function getByKeys(keys: string[]): any {
```
Comment (kabirbaidhya) [inline src/impl/domain.ts]: Generic function same as above. 
```suggestion
export function getByKeys<T>(keys: string[]): T[] {
```

File: src/impl/domain.ts [hunk 5]
```diff
@@ -118,6 +118,28 @@ export function get(key: string): any {
...
+ * Retrieves all values that correspond to a given list of keys.
+ *
+ * Example:
+ *  const a = 1;
+ *  const b = 2;
+ *  const sum = a + b;
+ *  store.set({a, b, sum})
+ *  const results = store.getByKeys(['a', 'b', 'sum']); // {a: 1, b: 2, sum: 3}
+ *
+ * @param {Array<string>} keys
```
Comment (kabirbaidhya) [inline src/impl/domain.ts]:
```suggestion
 * @param {string[]} keys
```

File: src/impl/domain.ts [hunk 6]
```diff
@@ -118,6 +118,28 @@ export function get(key: string): any {
...
+ * @returns {*}
+ */
+export function getByKeys(keys: string[]): any {
+  const result: any = {};
```
Comment (kabirbaidhya) [inline src/impl/domain.ts]: The implementation change - could be oneliner like this:
```ts
return keys.map(key => get(key));
```

File: test/domain.test.ts [hunk 7]
```diff
@@ -54,6 +54,21 @@ describe('store: [adapter=DOMAIN]', () => {
...
   });
 
+  describe('getByKeys()', () => {
+    it('should throw an error if store not initialized.', () => {
+      expect(globalStore.get.bind(globalStore, 'foo')).to.throw('No active domain found in store.');
+    });
+
+    it('should return an object with `undefined` as the value for the requested key if that key was not set.', done => {
+      const callback = () => {
+        expect(globalStore.getByKeys(['foo'])).to.deep.equal({ foo: undefined });
```
Comment (kabirbaidhya) [inline test/domain.test.ts]:
```suggestion
        expect(globalStore.getByKeys(['foo', 'bar'])).to.deep.equal([undefined, undefined]);
```

File: test/domain.test.ts [hunk 8]
```diff
@@ -54,6 +54,21 @@ describe('store: [adapter=DOMAIN]', () => {
...
+        done();
+      };
+
+      globalStore.initialize(adapter)(callback);
+    });
```
Comment (kabirbaidhya) [inline test/domain.test.ts]: Add one more test block here to test the positive case. For instance this:
```ts
expect(globalStore.getByKeys(['foo', 'bar', 'baz'])).to.deep.equal(['One', 'Two', 'Three']);
```

File: test/domain.test.ts [hunk 9]
```diff
@@ -266,7 +281,7 @@ describe('store: [adapter=DOMAIN]', () => {
 
         globalStore.set({ foo: 'Foo' });
 
-        expect(globalStore.get('foo')).to.equal('Foo');
+        expect(globalStore.getByKeys(['foo', 'bar'])).to.deep.equal({ foo: 'Foo', bar: 'bar' });
```
Comment (kabirbaidhya) [inline test/domain.test.ts]: Undo this change - as there are getByKeys tests above separately.

File: test/domain.test.ts [hunk 10]
```diff
@@ -283,7 +298,7 @@ describe('store: [adapter=DOMAIN]', () => {
             expect(globalStore.get('foo')).to.equal('bar');
           })
           .then(() => {
-            expect(globalStore.get('foo')).to.equal('bar');
+            expect(globalStore.getByKeys(['foo', 'bar'])).to.deep.equal({ foo: 'bar', bar: 'bonk' });
```
Comment (kabirbaidhya) [inline test/domain.test.ts]: Same here. Undo this.

**Comment by spencerwi** [comment]
Sure, I'll make those changes. I think I misunderstood the original issue.

In terms of destructuring, this is legal:

```js
// assuming the store has {a: 1, b: 2, sum: 3, empty: null}
let {a, b, sum, empty, notFoundThing} = store.getByKeys(['a', 'b', 'sum', 'empty', 'notFoundThing']);
console.log(a) // 1
console.log(b) // 2
console.log(sum) // 3
console.log(empty) // null
console.log(notFoundThing) // undefined
```

**Comment by spencerwi** [comment]
I've made the changes you requested -- thanks for the feedback! Hopefully the PR in its current state better fits the desired use-case. 😄
